core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -97,6 +97,7 @@ class KafkaApis(val requestChannel: RequestChannel,
         case requestId => throw new KafkaException("Unknown api code " + requestId)
       }
     } catch {
+      case fee: FatalExitError => throw fee
```

Comment:
Sorry for the delay in getting to this. I'm going to try and get this PR to the finish line (while still crediting @maysamyabandeh as the author since he did most of the work) and I was trying to understand the reason for this line. Under what scenario do we throw a `FatalExitException` that is then caught by this catch block? Rethrowing here prevents us from sending a response to the client, but it doesn't kill the server since `KafkaRequestHandler.run` also has a `catch: Throwable` block.